### PR TITLE
Add OP_RETURN into `ord wallet send`

### PR DESCRIPTION
**Pull request by grdddj**
Fixes https://github.com/ordinals/ord/issues/3007:
- adds a new `op-return` flag to `ord wallet send` that allows for sending some smaller data in `OP_RETURN`

Next steps/improvements:
- adding tests
- addressing TODOs

Possible next steps:
- support even multiple `OP_RETURN`s in one transaction (if possible)
- add `OP_RETURN` possibility even to the minting transaction

### CODE REVIEW

**Comment by raphjaph** [comment]
Let me know when this is ready for review!

**Comment by grdddj** [comment]
> Could you expand a bit on the use cases for this?

The use-case is when transferring the inscription between addresses, create some possibility to **add some metadata/info to the transfer** itself.

Concrete example:
- inscriptions are held by a service in one wallet, but "owned" by multiple customers of this service (on different addresses)
- the service allows for on-chain transfer of these inscriptions/certificates, so the transfer of ownership is transparent and verifiable
- `OP_RETURN` adds the nice possibility to specify e.g. the `price in BTC` that one user paid to buy the certificate from the other user
- the idea is that these OP-returns will be nicely visible e.g. on `mempool.space` (in the future maybe even in `ord` explorer) and will add more transparency to the transfer (or any other on-chain action)

image

Marked it as a draft for now, hopefully, I will have time to finish it soon. Thanks :)

File: src/subcommand/wallet/send.rs [hunk 1]
```diff
@@ -95,8 +100,15 @@ impl Send {
       change,
       self.fee_rate,
       postage,
-    )
-    .build_transaction()?;
+    );
+
+    if let Some(op_return) = self.op_return {
+      // TODO: improve the check for checking byte-length
+      assert!(op_return.len() <= 80);
```
Comment (raphjaph) [inline src/subcommand/wallet/send.rs:107]:
```suggestion
      assert!(op_return.len() <= 82);
```

File: src/wallet/transaction_builder.rs [hunk 2]
```diff
@@ -36,6 +36,8 @@ use {
   std::cmp::{max, min},
 };
 
+use bitcoin::blockdata::script::PushBytesBuf;
+
```
Comment (raphjaph) [inline src/wallet/transaction_builder.rs:40]: Inline with the other imports

**Comment by raphjaph** [comment]
I'm going to close this but please reopen if you wish to continue work on this!

**Pull request closed without merging**